### DISCUSSION TITLE
supervision-0.15.0

### DISCUSSION THREAD

**Discussion by SkalskiP** [roboflow/supervision#413] [Announcements]
## 🚀 Added

https://github.com/roboflow/supervision/assets/26109316/4d6c4a70-b40e-48fc-9e58-23b7e67bf94a

- [`sv.LabelAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.LabelAnnotator) allowing to annotate images and videos with text. ([#170](https://github.com/roboflow/supervision/pull/170))

- [`sv.BoundingBoxAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.BoundingBoxAnnotator) allowing to annotate images and videos with bounding boxes. ([#170](https://github.com/roboflow/supervision/pull/170))

- [`sv.BoxCornerAnnotator `](https://supervision.roboflow.com/annotators/#supervision.annotators.core.BoxCornerAnnotator) allowing to annotate images and videos with just bounding box corners. ([#170](https://github.com/roboflow/supervision/pull/170))

- [`sv.MaskAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.MaskAnnotator) allowing to annotate images and videos with segmentation masks. ([#170](https://github.com/roboflow/supervision/pull/170))

- [`sv.EllipseAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.EllipseAnnotator) allowing to annotate images and videos with ellipses (sports game style). ([#170](https://github.com/roboflow/supervision/pull/170))

- [`sv.CircleAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.CircleAnnotator) allowing to annotate images and videos with circles. ([#386](https://github.com/roboflow/supervision/pull/386))

- [`sv.TraceAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.TraceAnnotator) allowing to draw path of moving objects on videos. ([#354](https://github.com/roboflow/supervision/pull/354)) 

- [`sv.BlurAnnotator`](https://supervision.roboflow.com/annotators/#supervision.annotators.core.BlurAnnotator) allowing to blur objects on images and videos. ([#405](https://github.com/roboflow/supervision/pull/405)) 

```python
>>> import supervision as sv

>>> image = ...
>>> detections = sv.Detections(...)

>>> bounding_box_annotator = sv.BoundingBoxAnnotator()
>>> annotated_frame = bounding_box_annotator.annotate(
...     scene=image.copy(),
...     detections=detections
... )
```

- Supervision usage [example](https://github.com/roboflow/supervision/tree/develop/examples/traffic_analysis). You can now learn how to perform traffic flow analysis with Supervision. ([#354](https://github.com/roboflow/supervision/pull/354)) 

https://github.com/roboflow/supervision/assets/26109316/c9436828-9fbf-4c25-ae8c-60e9c81b3900

## 🌱 Changed

- [`sv.Detections.from_roboflow`](https://supervision.roboflow.com/detection/core/#supervision.detection.core.Detections.from_roboflow) now does not require `class_list` to be specified. The `class_id` value can be extracted directly from the [inference](https://github.com/roboflow/inference) response. ([#399](https://github.com/roboflow/supervision/pull/399))

- [`sv.VideoSink`](https://supervision.roboflow.com/utils/video/#videosink) now allows to customize the output codec. ([#381](https://github.com/roboflow/supervision/pull/381))

- [`sv.InferenceSlicer`](https://supervision.roboflow.com/detection/tools/inference_slicer/#supervision.detection.tools.inference_slicer.InferenceSlicer) can now operate in multithreading mode. ([#361](https://github.com/roboflow/supervision/pull/361))

## 🛠️ Fixed

- [`sv.Detections.from_deepsparse`](https://supervision.roboflow.com/detection/core/#supervision.detection.core.Detections.from_deepsparse) to allow processing empty [deepsparse](https://github.com/neuralmagic/deepsparse) result object. ([#348](https://github.com/roboflow/supervision/pull/348))

## 🏆 Contributors

@hardikdava (Hardik Dava), @onuralpszr (Onuralp SEZER), @Killua7362 (Akshay Bhat), @fcakyon (Fatih C. Akyon), @akashAD98 (Akash A Desai), @Rajarshi-Misra, @capjamesg (James Gallagher), @SkalskiP (Piotr Skalski)

<hr /><em>This discussion was created from the release <a href='https://github.com/roboflow/supervision/releases/tag/0.15.0'>supervision-0.15.0</a>.</em>

**Comment by zentoliyan** [comment]
oh gosh, that's fantastic :))

**Reply by SkalskiP** [reply]
Best part is we already have 2 more Annotators waiting for next release! 🔥

**Reply by AlainPilon** [reply]
Are you planning on a fixed release schedule or just release whenever there is enough new features?

**Reply by SkalskiP** [reply]
Hi, @AlainPilon 👋🏻 Up until now, we were released when we had enough new features. And I feel we will continue to work this way. I don't want to force release when we have nothing new to show. 

I usually communicate a week before through Discussions. Now we also have Discord so I'll let people now there as well.

**Comment by AlainPilon** [comment]
>  Now we also have Discord so I'll let people now there as well.

Where is the link?

**Reply by capjamesg** [reply]
Here you go! https://discord.com/invite/GbfgXGJ8Bk

**Reply by SkalskiP** [reply]
Thanks @capjamesg 🙏🏻! The link is at the top of the README.md as well.
